lib/semantics/check-call.cpp

```diff
@@ -610,11 +610,24 @@ static void CheckExplicitInterfaceArg(evaluate::ActualArgument &arg,
                 messages.Say(
                     "Actual argument is not a variable or typed expression"_err_en_US);
               }
-            } else if (!object.type.type().IsAssumedType()) {
+            } else {
               const Symbol &assumed{DEREF(arg.GetAssumedTypeDummy())};
-              messages.Say(
-                  "Assumed-type TYPE(*) '%s' may be associated only with an assumed-TYPE(*) %s"_err_en_US,
-                  assumed.name(), dummyName);
+              if (!object.type.type().IsAssumedType()) {
+                messages.Say(
+                    "Assumed-type TYPE(*) '%s' may be associated only with an"
+                    " assumed-TYPE(*) %s"_err_en_US,
+                    assumed.name(), dummyName);
+              } else if (const auto *details{
+                             assumed
+                                 .detailsIf<ObjectEntityDetails>()}) {  // C711
```

Comment:
If you could find a different place for the comment (e.g. on the Say statement), clang-format wouldn't make such a big mess of this.

Reply:
Thanks, Tim.  I'll move it.